File: exercicios/codigosEmOutrasLinguagens/EX5FU/EX5FUv2.c
#include <stdio.h>

void decimalParaBinario(int num) {
    int binario[32];
    int indice = 0;
    int i;

    if (num == 0) {
        printf("Binário: 0\n");
        return;
    }

    while (num > 0) {
        binario[indice] = num % 2;
        num = num / 2;
        indice++;
    }

    printf("Binário: ");

    for (i = indice - 1; i >= 0; i--) {
        printf("%d", binario[i]);
    }
    
    printf("\n");
}

int main() {
    int num;

    printf("Digite o 1º número inteiro positivo: ");
    scanf("%d", &num);
    decimalParaBinario(num);

    printf("Digite o 2º número inteiro positivo: ");
    scanf("%d", &num);
    decimalParaBinario(num);

    printf("Digite o 3º número inteiro positivo: ");
    scanf("%d", &num);
    decimalParaBinario(num);

    return 0;
}

File: exercicios/codigosEmOutrasLinguagens/EX5FU/EX5FUv2.py
def decimalParaBinario(num):
    binario = []

    if num == 0:
        print("Binário: 0")
        return

    while num > 0:
        binario.append(num % 2)
        num = num // 2

    print("Binário:", end=" ")
    
    for i in range(len(binario) - 1, -1, -1):
        print(binario[i], end="")
        
    print()

num = int(input("Digite o 1º número inteiro positivo: "))
decimalParaBinario(num)

num = int(input("Digite o 2º número inteiro positivo: "))
decimalParaBinario(num)

num = int(input("Digite o 3º número inteiro positivo: "))
decimalParaBinario(num)